core/shared/src/main/scala/fs2/internal/SyncRef.scala

```diff
@@ -0,0 +1,65 @@
+package fs2.internal
+
+import java.util.concurrent.atomic.AtomicReference
+
+import cats.effect.Sync
+import fs2.async.Ref.Change
+
+import scala.annotation.tailrec
+
+/**
+  * Reference to `A` that supports concurrent, but only synchronous operations.
+  *
+  * Think of it as an java AtomicReference wrapped in `F`
+  *
+  */
+final class SyncRef[F[_], A] (private val ar: AtomicReference[A])(implicit F: Sync[F]) {
```

Comment:
I've wanted this type a number of times and I've seen people re-invent it. Perhaps we should consider putting a more polished version of it in `fs2.async`.

Reply:
yeah, is a bit of awkward to put in async :-) beacuse it is sync, but I may just add it there and see if that works ?